### PR TITLE
adding parallel execution to restyle-diff

### PR DESCRIPTION
**Pull request by Alami-Amine**
- calling function that runs container using `xargs`, with 256 Maximum number of arguments
- add option to pull new restyled image
-  added a git fetch upstream

### CODE REVIEW

File: scripts/helpers/restyle-diff.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
 #
-# Copyright (c) 2020 Project CHIP Authors
+# Copyright (c) 2024 Project CHIP Authors
```
Comment (cecille) [inline scripts/helpers/restyle-diff.sh:4]:
```suggestion
# Copyright (c) 2020-2024 Project CHIP Authors
```

File: scripts/helpers/restyle-diff.sh [hunk 2]
```diff
@@ -45,33 +46,67 @@ docker_run() {
 }
 
 restyle-paths() {
-
     image=restyled/restyler:edge
+    batch_size=4
```
Comment (andy31415) [inline scripts/helpers/restyle-diff.sh:50]: How about using `$(nproc)`? or is that too much?

Do these restylers support multiple paths in some way?

Comment (andy31415) [reply scripts/helpers/restyle-diff.sh:50]: I tried locally just passing in all files and it seemed to work. The worst may be if we somehow have too many files changed (regen?) and they do not fit in max command line arguments. We should either just pass all (first try) or pass in large batches that are configurable (like 256 or 1024 files at once or something)

Comment (tcarmelveilleux) [reply scripts/helpers/restyle-diff.sh:50]: Is there a way to get command line arguments from a file? I would think so.

BTW on linux, max command line length is ~100KiB

Comment (andy31415) [reply scripts/helpers/restyle-diff.sh:50]: we can xargs it with a large count like 256 and just assume it is ok. My `getconf ARG_MAX` on my machine is about 2MB. Or even 100 files at a time will make most cases just be ok, even the codegen ones.

Comment (Alami-Amine) [reply scripts/helpers/restyle-diff.sh]: ok i just added xargs, makes the script much more concise...

Comment (Alami-Amine) [reply scripts/helpers/restyle-diff.sh]: i made it accept 256, tested it with some old branch and it isnt crashing so looks good

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Pull request merged**